src/awkward/_broadcasting.py

```diff
@@ -670,39 +670,55 @@ def continuation():
                 isinstance(x, RegularArray) or not isinstance(x, listtypes)
                 for x in inputs
             ):
-                maxsize = max(x.size for x in inputs if isinstance(x, RegularArray))
+                # Ensure all layouts have same length
+                length = None
+                for x in inputs:
+                    if isinstance(x, Content):
+                        if length is None:
+                            length = x.length
+                        elif backend.nplike.known_shape:
+                            assert length == x.length
+                assert length is not None
+
+                if any(x.size == 0 for x in inputs if isinstance(x, RegularArray)):
+                    dimsize = 0
```

Comment:
Right, the original motivation for the `max` was because the target size is "the one that isn't `1`." But if there's any `0`, then it broadcasts to `0`.